Packs/AppSentinelsAi/Integrations/AppSentinelsAi/AppSentinelsAi.py

```diff
@@ -0,0 +1,347 @@
+BASE_REQUEST_PARAMS: dict = {"page": "0", "limit": "1000", "sort": "timestamp", "sort_by": "asc", "include_system": "false"}
+
+""" CLIENT CLASS """
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API"""
+
+    def __init__(
+        self,
+        base_url: str,
+        user_key: str,
+        api_key: str,
+        organization: str,
+        base_request_body: dict,
+        base_request_params: dict,
+        verify: bool,
+        use_proxy: bool,
+    ) -> None:
+        """
+        Prepare constructor for Client class.
+
+        Calls the constructor of BaseClient class and updates the header with the authentication token.
+
+        Args:
+            base_url: The url of AppSentinels.ai instance.
+            user_key: The user key for AppSentinels.ai API.
+            api_key: The Api key for AppSentinels.ai API - specific for every licensing.
+            organization: The organization ID for AppSentinels.ai API.
+            base_request_body: The base body for the http request.
+            base_request_params: The base params for the http request.
+            verify: True if verify SSL certificate is checked in integration configuration, False otherwise.
+            use_proxy: True if the proxy server needs to be used, False otherwise.
+        """
+
+        super().__init__(base_url=base_url, verify=verify, proxy=use_proxy)
+        self._headers = {
+            "accept": "application/json",
+            "apikey": api_key,
+            "x-user-key": user_key,
+            "Content-Type": "application/json",
+        }
+        self.organization = organization
+        self.base_request_body = base_request_body
+        self.base_request_params = base_request_params
+        self.api_key = api_key
+        self.user_key = user_key
+
+    def get_events_request(self, params_update: dict, body_update: dict) -> dict:
+        """
+        Retrieve the detections from AppSentinels.ai  API.
+
+        Args:
+            params_update (dict): The param update to add to the base params.
+            body_update (dict): The body update to add to the base body.
+
+        """
+        url_suffix = f"/api/v1/{self.organization}/audit-logs"
+        params = self.base_request_params.copy()
+        params.update(params_update)
+        body = self.base_request_body.copy()
+        body.update(body_update)
+        return self._http_request(
+            "POST", url_suffix=url_suffix, headers=self._headers, json_data=body, params=params, resp_type="json"
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def remove_first_run_params(params: dict[str, Any]) -> None:
+    """
+    Remove the "First Run" items from the param dictionary.
+
+    Args:
+        params (Dict[str, Any]): Integration parameters.
+
+    """
+    for key in ("from_date", "to_date"):
+        params.pop(key, None)
+
+
+def fetch_events_list(client: Client, last_run: Dict, fetch_limit: int | None, use_last_run_as_body: bool) -> List[Dict]:
```

Comment:
This function is a bit messy and unclear.
Let's try to simplify it 
Reduce nested ifs, remove redundant checks etc. 
Like: if not response_data or not response_data.get("data"): → replaced with data = response.get("data") if response else []